pkg/registry/external_metrics/reststorage.go

```diff
@@ -28,23 +28,26 @@ import (
 	"k8s.io/apiserver/pkg/registry/rest"
 	"k8s.io/metrics/pkg/apis/external_metrics"
 
+	"sigs.k8s.io/custom-metrics-apiserver/pkg/apiserver/metrics"
 	"sigs.k8s.io/custom-metrics-apiserver/pkg/provider"
 )
 
 // REST is a wrapper for CustomMetricsProvider that provides implementation for Storage and Lister
 // interfaces.
 type REST struct {
-	emProvider provider.ExternalMetricsProvider
+	emProvider        provider.ExternalMetricsProvider
+	freshnessObserver metrics.FreshnessObserver
 	rest.TableConvertor
 }
 
 var _ rest.Storage = &REST{}
 var _ rest.Lister = &REST{}
 
 // NewREST returns new REST object for provided CustomMetricsProvider.
-func NewREST(emProvider provider.ExternalMetricsProvider) *REST {
+func NewREST(emProvider provider.ExternalMetricsProvider, freshnessObserver metrics.FreshnessObserver) *REST {
```

Comment:
Since each provider has its own NewRest function, we could deduce the apigroup and not require users to pass the FreshnessObserver themselves.

Reply:
:+1: